### ISSUE TITLE
Automatic installation is required

### ISSUE THREAD

**Issue by gtaifu** [DiCarloLab-Delft/PycQED_py3#427]
## Automation
Current PycQED installation is a bit time-consuming and requires support from an experienced guy, especially installing dependency packages. I started automating the installation process of PycQED by introducing the PowerShell script (`install.ps1`) for Windows OS. Recently, I learnt that the following steps can be further automated:
- Python package installation can be done via `pip install -r requirements.txt`. All dependency packages can be listed in the file `requirements.txt`, which enables an easy package control.
- Terminal scripts should be provided for Windows, Linux, and iOS. A first version for Windows OS (`install.ps1`) is in place. We need to update it and also provide the Linux version.
- Update the installation manual to adapt to the new installation process.

## Conda Environment Issue
Current PycQED requires Python 3 and several packages. It directly runs in the main environment of the PC.  If the PC is also used for other projects and may require some specific packages or specific version of some package, the installation of PycQED may damage the original environment and might be a bit aggressive to the original PC environment. 

To avoid this problem, Anaconda introduced the concept [virtual environments](https://conda.io/docs/user-guide/tasks/manage-environments.html). A virtual environment can be used for a dedicated projects. In different virtual environments, different Python versions, Python packages and  package versions can be used, which are completely independent of other environments. For example, Alice originally uses her computer for a project that requires Python 2. One day, she entered the wonderful world of PycQED. If she directly install PycQED which uses Python 3, then the installation will break her original environment, which is not desired. However, by using Anaconda virtual environments, she can happily install PycQED without worrying about breaking her original experiment environment.

I suggest we refine the installation process to enable the user to select the following options:
- Install PycQED in my default environment, because I use PycQED everyday and it is my main experiment environment.
- Install PycQED into a virutal environment called pycqed, since I also work with other projects and I do not want PycQED to break my original environment.

What do you think about it? @AdriaanRol @cdickel @elrama-

**Comment by elrama-** [comment]
I like this proposal very much. Let's work on it once I get back to delft

**Comment by gtaifu** [comment]
It seems the package `qutip` is not required according to the [installation instructions](https://github.com/DiCarloLab-Delft/PycQED_py3/blob/master/docs/install.md), but it is written down in the [requirements.txt](https://github.com/DiCarloLab-Delft/PycQED_py3/blob/master/requirements.txt). 

Which module requires it?

**Comment by AdriaanRol** [comment]
It's being used quite extensively in plotting density matrices for e.g.,
the Tomography analysis.

Adriaan Rol

T:    +316 29 470 740
M:  adriaan.rol@gmail.com

**Comment by gtaifu** [comment]
@AdriaanRol 
Regarding the issue "unable to find vcvarsall.bat" as mentioned in the file [compiling_with_cython.md](https://github.com/DiCarloLab-Delft/PycQED_py3/blob/master/docs/compiling_with_cython.md), I cannot repeat this error on my computer. 

I am wondering if it is because I have already enabled `vcvarsall.bat` in the PowerShell in the correct way or not? By correct way, I mean the following commands are put in my PowerShell profile which is invoked when launching a PowerShell terminal:
```
$Env:VS90COMNTOOLS= $Env:VS140COMNTOOLS
$VS_bat_path = "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvarsall.bat"
Invoke-BatchFile $VS_bat_path amd64
```
If this is the case, then the above lines are the solution to the Cython installation problem.

**Comment by gtaifu** [comment]
@AdriaanRol  As you mentioned, installing the package `pygraphviz` would be a challenge. I failed to install this package by plainly using `pip install pygraphviz`.

Is this [link](https://stackoverflow.com/questions/40809758/howto-install-pygraphviz-on-windows-10-64bit) containing the solution to the challenge you mentioned to me?

**Comment by AdriaanRol** [comment]
@gtaifu , the link you post may very well contain the solution. 
However, when I went through all that digging, I wrote down the links I used and the steps taken. This is shown [in the installation instructions of autodepgraph](https://github.com/AdriaanRol/AutoDepGraph#installation). 

It will probably link to the same post but also contains some other hints on how to do it. If you find anything to be different, let me know and we can edit my instructions. I also know that @elrama- has previously successfully installed this on windows.
